### PR TITLE
MVI upgrade to versionCode 4.1

### PR DESCRIPTION
**Pull request by jholton**
## Description of change
Change `versionCode` specified in MVI requests from 3.0 to 4.1.
Re-ran specs and updated VCR cassettes to identify any significant changes in responses.

## Testing done
Re-ran specs and updated cassettes.

#### Unique to this PR

- [x] Change `versionCode` specified in MVI requests from 3.0 to 4.1.
- [x] Updated specs and VCR cassettes

#### Applies to all PRs

- [x] Appropriate logging
- [ ] ~~Swagger docs have been updated, if applicable~~
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: spec/lib/mvi/service_spec.rb [hunk 1]
```diff
@@ -81,7 +81,7 @@
         VCR.use_cassette('mvi/find_candidate/valid_vet360_id') do
           response = subject.find_profile(user)
           expect(response.status).to eq('OK')
-          expect(response.profile['vet360_id']).to eq('123456789')
+          expect(response.profile['vet360_id']).to eq('1000')
```
Comment (jholton) [inline spec/lib/mvi/service_spec.rb]: Vet360 ID is parsed [here](https://github.com/department-of-veterans-affairs/vets-api/blob/90970fd0e4504b66e70358bee56fc49309d13f59/lib/mvi/responses/id_parser.rb#L34).
`vet360_id: select_ids(select_extension(ids, /^\w+\^PI\^200VETS\^USDVA\^\w+$/, CORRELATION_ROOT_ID))&.first`

You can see that is just taking the first found result.
In our old sample response, there was just one matching Vet360 ID.
In the new response there are multiple Vet360 IDs.
Is just taking the first the best idea still?  Should all of them be returned by the parser instead (like `mhv_ids`, `vha_facility_ids`, etc)? 

I added check marks in the code below to indicate matching rows.

Old response:
```
<subject1 typeCode="SBJ">
    <patient classCode="PAT">
        <id extension="1008714701V416111^NI^200M^USVHA^P" root="2.16.840.1.113883.4.349"/>
        √ <id extension="123456789^PI^200VETS^USDVA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="796122306^PI^200BRLS^USVBA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="9100792239^PI^200CORP^USVBA^A" root="2.16.840.1.113883.4.349"/>
```

New response:
```
<subject1 typeCode="SBJ">
    <patient classCode="PAT">
        <id extension="1008787551V609092^NI^200M^USVHA^P" root="2.16.840.1.113883.4.349"/>
        <id extension="768598574^PI^200BRLS^USVBA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="0000001008787551V609092000000^PI^200ESR^USVHA^A" root="2.16.840.1.113883.4.349"/>
        √ <id extension="1000^PI^200VETS^USDVA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="201801104^PI^523^USVHA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="22710152^PI^585^USVHA^A" root="2.16.840.1.113883.4.349"/>
        √ <id extension="2011801312^PI^200VETS^USDVA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="32364231^PI^200CORP^USVBA^A" root="2.16.840.1.113883.4.349"/>
        √ <id extension="201808031^PI^200VETS^USDVA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="201804032^PI^200VETS^USDVA^H" root="2.16.840.1.113883.4.349"/>
        <id extension="7174051^PI^608^USVHA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="0000001008787550V443247000000^PI^200ESR^USVHA^H" root="2.16.840.1.113883.4.349"/>
        <id extension="875637483^PI^200BRLS^USVBA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="32364195^PI^200CORP^USVBA^A" root="2.16.840.1.113883.4.349"/>
        √ <id extension="80^PI^200VETS^USDVA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="201806031^PI^500^USVHA^A" root="2.16.840.1.113883.4.349"/>
        <id extension="201806032^PI^500^USVHA^A" root="2.16.840.1.113883.4.349"/>
```

Comment (annaswims) [reply spec/lib/mvi/service_spec.rb]: oof, I don't think we're equipped to handle multiple vets360 id's properly.  I think this version upgrade was supposed to give us additional status codes (id's that end in `^L`).  Maybe we need to reach out to the MVI or Vets360 teams to see how to properly handle.

https://dsva.slack.com/archives/CK1FA11H8/p1565638882180700

**Comment by annaswims** [comment]
One of the main things I'm wondering is how much of this data change is because of the version change and how much is because the data in MVI changes over time.

File: lib/mvi/responses/id_parser.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@ def parse(ids)
                                          %w[H L])&.first,
           vha_facility_ids: select_facilities(select_extension(ids, /^\w+\^PI\^\w+\^USVHA\^\w+$/, CORRELATION_ROOT_ID)),
           birls_id: select_ids(select_extension(ids, /^\w+\^PI\^200BRLS\^USVBA\^\w+$/, CORRELATION_ROOT_ID))&.first,
-          vet360_id: select_ids(select_extension(ids, /^\w+\^PI\^200VETS\^USDVA\^\w+$/, CORRELATION_ROOT_ID))&.first,
+          vet360_id: select_ids(select_extension(ids, /^\w+\^PI\^200VETS\^USDVA\^A$/, CORRELATION_ROOT_ID))&.first,
```
Comment (jholton) [inline lib/mvi/responses/id_parser.rb]: There can be historical Vet360 IDs.  Adding the `A` flag requirement to ensure only an active ID is selected.

File: spec/lib/mvi/service_spec.rb [hunk 3]
```diff
@@ -249,8 +247,6 @@
 
     context 'when a MVI invalid request response is returned' do
       it 'should raise a invalid request error', :aggregate_failures do
-        invalid_xml = File.read('spec/support/mvi/find_candidate_invalid_request.xml')
-        allow_any_instance_of(MVI::Service).to receive(:create_profile_message).and_return(invalid_xml)
```
Comment (jholton) [inline spec/lib/mvi/service_spec.rb]: This uses a VCR cassette, which records the request and the response. Therefore these two lines are are unnecessary.  Also removed the `find_candidate_invalid_request.xml` file

File: spec/lib/mvi/service_spec.rb [hunk 4]
```diff
@@ -262,21 +258,6 @@
       end
     end
 
-    context 'when a MVI failure response is returned' do
-      it 'should raise a request failure error', :aggregate_failures do
```
Comment (jholton) [inline spec/lib/mvi/service_spec.rb]: This is a duplicate of the preceding spec.  
It is built with the same request XML, so the results will be exactly the same.

File: spec/lib/mvi/service_spec.rb (hunk 1 above)
Comment (jholton) [reply spec/lib/mvi/service_spec.rb]: Confirmed with Vet360 that a user should only have one active (`A`) id.
MVI changed the in this `dev` environment for this user to reflect this.
We _shouldn't_ run into this issue in production.

**Comment by jholton** [comment]
> can you provide some context as to why this is needed?

Yup.  [Here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/462) is the motivating issue.
We need to make sure we are handling the various `IDSTATUS` values correctly, including an `IDSTATUS` of `L`.  `L` was added in MVI version 4.1, hence the need for the upgrade.

File: spec/support/mvi/find_candidate_failure_response.xml [hunk 5]
```diff
@@ -1,20 +1,26 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!--
+  This is a legacy response.
+  It is not representative of an actual response that would contain an 'AR' typeCode.
+  This response (INVALID REQUEST) today (Sept 12, 2019) would contain an 'AE' typeCode.
+  This will be used to test processing 'AR' responses until we can generate a more realistic response.
```
Comment (kfrz) [inline spec/support/mvi/find_candidate_failure_response.xml]: What's blocking us from generating a  

> more realistic response

**Comment by annaswims** [comment]
Just a couple failing specs in modules/openid_auth/spec/requests/mvi_user_request_spec.rb

File: spec/support/mvi/find_candidate_failure_response.xml (hunk 5 above)
Comment (jholton) [reply spec/support/mvi/find_candidate_failure_response.xml]: 'AR' is returned when something is wrong with the MVI service
> AR - fail to process (reject) the message for reasons unrelated to its content or format (system down, internal error, etc.).  For most such problems it is likely that the responding system will be able to accept the same message at a later time.

I can't send a request that causes the MVI backend to simulate a system problem.

**Pull request merged**